core/src/main/scala/org/apache/spark/deploy/Client.scala

```diff
@@ -92,14 +102,15 @@ private class ClientActor(driverArgs: ClientArguments, conf: SparkConf)
     println("... waiting before polling master for driver state")
     Thread.sleep(5000)
     println("... polling master for driver state")
-    val statusFuture = (masterActor ? RequestDriverStatus(driverId))(timeout)
+    val statusFuture = (activeMasterActor ? RequestDriverStatus(driverId))(timeout)
       .mapTo[DriverStatusResponse]
     val statusResponse = Await.result(statusFuture, timeout)
-
     statusResponse.found match {
       case false =>
-        println(s"ERROR: Cluster master did not recognize $driverId")
-        System.exit(-1)
+        statusResponse.exception.getOrElse {
+          println(s"ERROR: Cluster master did not recognize $driverId")
+          System.exit(-1)
+        }
```

Comment:
This changes behavior. If `statusResponse.exception` is defined then this doesn't exit. I would just revert this change.